docs/guide/features.md

```diff
@@ -368,6 +370,38 @@ By default, the worker script will be emitted as a separate chunk in the product
 import MyWorker from './worker?worker&inline'
 ```
 
+### new Worker()
+
+A web worker script can be imported by `new Worker()`. The difference between this syntax and imported by appending `?worker` or `?sharedworker` is that this one can use worker options. So we can create a `classic worker` by this syntax.
+
+Detect the case where `{ type: 'classic' }` is not provided or `{ type: 'classic' }` is passed with a static string and use classic workers.
+
+```ts
+const worker = new Worker(new URL('./worker.js', import.meta.url), {
+  type: 'classic'
+})
+const worker = new SharedWorker(new URL('./worker.js', import.meta.url), {
+  type: 'classic'
+})
+```
+
+::: tip Note
+
+the worker options must be static for Vite to analyze it. if it contains dynamic expressions, `/* @vite-ignore */` can be used to opt-out of Vite processing
+
+```ts
+const generateWorkerOption = () => ({ type: 'classic' })
+// use dy dynamic expression will got a vite process error.
+const worker = new Worker(
+  new URL('./worker.js', import.meta.url),
+  generateWorkerOption()
+)
+```
+
+No matter which type of worker is used, the same [worker configuration](/config/#worker-options) will be used for bundle.
+
+:::
```

Comment:
I think we can skip the `vite-ignore` part here, I wonder what others think about this. Because it's an edge case and if Vite hits something like this, the dev server would already suggest using `@vite-ignore`.


If we do remove that, we can simplify this entire part to explain about the worker configuration.

```suggestion
See [Worker Options](/config/#worker-options) for details on configuring the bundling of all workers.
```

Reply:
good point.